pulsar-client/src/main/java/org/apache/pulsar/client/impl/ZeroQueueConsumerImpl.java

```diff
@@ -182,14 +182,20 @@ void receiveIndividualMessagesFromBatch(BrokerEntryMetadata brokerEntryMetadata,
         log.warn(
                 "Closing consumer [{}]-[{}] due to unsupported received batch-message with zero receiver queue size",
                 subscription, consumerName);
-        // close connection
-        closeAsync().handle((ok, e) -> {
```

Comment:
The `closeAsync()` method already fails all messages in the `pendingReceives` queue. As such, the callback on the `closeAsync()` future hasn't ever had any affect. I think it is a good catch to re-order these tasks. However, I wonder if there are any chances for race conditions because we're delaying putting the consumer into a `Closing` state.

Reply:
This changes not about race conditions. It more like unify code. notifyPendingReceivedCallback should in  internalPinnedExecutor.
And fix the bad behavior: close consumer firstly, makes pendings receives get inaccurate error info.